src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php

```diff
@@ -129,18 +130,46 @@ public function reverseTransform($value)
             $value = str_replace(',', $decSep, $value);
         }
 
+        if (false !== strpos($value, $decSep)) {
+            $type = \NumberFormatter::TYPE_DOUBLE;
+        } else {
+            $type = PHP_INT_SIZE === 8 ? \NumberFormatter::TYPE_INT64 : \NumberFormatter::TYPE_INT32;
+        }
+
         // replace normal spaces so that the formatter can read them
-        $value = $formatter->parse(str_replace(' ', "\xc2\xa0", $value));
+        $result = $formatter->parse(str_replace(' ', "\xc2\xa0", $value), $type, $position);
 
         if (intl_is_failure($formatter->getErrorCode())) {
             throw new TransformationFailedException($formatter->getErrorMessage());
         }
 
         if (self::FRACTIONAL == $this->type) {
-            $value /= 100;
+            $result /= 100;
         }
 
-        return $value;
+        if (function_exists('mb_detect_encoding') && false !== $encoding = mb_detect_encoding($value, null, true)) {
+            $length = mb_strlen($value, $encoding);
```

Comment:
Are you sure about it ? `NumberFormatter::parse` is documented as setting the offset in the string on php.net. This offset will be based on bytes, not on characters, so you should not be using `mb_strlen`

Reply:
Nope, I just adapted the code from the `NumberToLocalizedStringTransformer`. Seems this was done on purpose in #7902.

Reply:
I looked at the tests for the `NumberToLocalizedStringTransformer` and copied over some of them. They prove that we indeed need to use the mbstring extension (for example, the exception message in `testReverseTransformDisallowsCenteredExtraCharactersMultibyte()` would not match otherwise).